File: packages/cli/src/cli/env.ts
export type EnvSource = Record<string, string | undefined>;

export const readEnvString = ({
  env,
  name,
}: {
  readonly env: EnvSource;
  readonly name: string;
}): string | undefined => {
  const value = env[name]?.trim();
  return value ? value : undefined;
};

export const readEnvList = ({
  env,
  name,
}: {
  readonly env: EnvSource;
  readonly name: string;
}): readonly string[] | undefined => {
  const value = readEnvString({ env, name });
  if (!value) {
    return undefined;
  }

  const items = value
    .split(",")
    .map((item) => item.trim())
    .filter((item) => item.length > 0);

  return items.length > 0 ? items : undefined;
};

export const preferFlag = <T>(
  flagValue: T | undefined,
  envValue: T | undefined,
): T | undefined => {
  return flagValue === undefined ? envValue : flagValue;
};

File: lib/src/cli/env.dart
import 'package:cliweave/cliweave.dart';

String? readEnvString(CommandContext context, String name) {
  final value = context.process.readEnv(name)?.trim();
  return value == null || value.isEmpty ? null : value;
}

List<String>? readEnvList(CommandContext context, String name) {
  final value = readEnvString(context, name);
  if (value == null) {
    return null;
  }
  final items = value
      .split(',')
      .map((item) => item.trim())
      .where((item) => item.isNotEmpty)
      .toList();
  return items.isEmpty ? null : items;
}

T? preferFlag<T>(T? flag, T? environment) => flag ?? environment;
